docs/release.md

```diff
@@ -74,23 +74,33 @@ It is possible that a core approver isn't a contrib approver. In that case, the
 ## Producing the artifacts
 The last step of the release process creates artifacts for the new version of the collector and publishes images to Dockerhub. The steps in this portion of the release are done in the [opentelemetry-collector-releases](https://github.com/open-telemetry/opentelemetry-collector-releases) repo.
 
-1. Update the `./distributions/**/manifest.yaml` files to include the new release version.
+**NOTE: Step 1 of this section is updated!**
```

Comment:
Do we need this note? I don't think people are able to see the old section when they go to this page

Reply:
That's fair just wanted to reduce the chance of causing someone trouble by making changes to the release process, since making changes to release process was a sticking point last time.